### DISCUSSION TITLE
Cell self-penetrations

### DISCUSSION THREAD

**Discussion by mojtabaAmiraslanpour** [UvaCsl/HemoCell#113] [General discussion]
Hi,
I am running pipeflow case for various Re numbers and I see sel-penetrations in cells in cases with moderate shear rates. Is this a normal behavior in HemoCell? I am using the standard material model for RBCs.

![Screenshot 2023-11-16 233154](https://github.com/UvaCsl/HemoCell/assets/15649053/dc798254-95d7-48a4-a290-651a0878315e)

Any hints are appreciated,
Thanks.

**Comment by ErenC1** [comment]
Hi,

I see similar behavior in my simulations and it increases with the shear. So, following this post.

**Comment by ErenC1** [comment]
I was reading Cimrak's book yesterday, it mentions that repulsive force between the particles of the "individual" RBC can prevent the folding. We already have repulsion in HemoCell (can be used as starting point), we just need to do little bit coding to define the force in each RBC. Right now I don't have time to work on this but we can keep in touch and fix this ourselves.

**Reply by mojtabaAmiraslanpour** [reply]
Can you please tell me which page is this discussed?

**Comment by ErenC1** [comment]
I would like to make another contribution with these intresting visuals. I have small voxels and repulsion cutoff between wall-cell is 1.5x of the dx. Still these kind of instabilities happens once in a while.

![Screenshot from 2023-11-27 22-03-18](https://github.com/UvaCsl/HemoCell/assets/57329486/fb59a86d-f290-4e99-a511-486c5aad4341)
![Screenshot from 2023-11-27 22-03-43](https://github.com/UvaCsl/HemoCell/assets/57329486/a896b234-cbf5-4c00-aca8-1ebecf8895d4)

**Reply by mojtabaAmiraslanpour** [reply]
What dx are you using for this problem?

**Comment by mojtabaAmiraslanpour** [comment]
Hi Eren,

It definitely would be nice to work on this together. I am currently trying various material models to see if I can get fewer self-penetrations while preserving the natural behavior of the membrane. One thing that I can say is that this only happens at high shear rates (i.e. 5000 1/s). I don's see such penetrations in low shear rates.

I haven't yet activated the wall-cell repulsive force. That is another thing that we can take a look. I am also interested in the narrow slit example and how this force can relate.

I also plan to understand better the repulsion force and how it is defined. Thank you for mentioning Cimrak's book. I hadn't seen it before. It seems it has many things covered.
